### ISSUE TITLE
CRD Manifests Are Not Ordered

### ISSUE THREAD

**Issue by akefirad** [cdk8s-team/cdk8s#1103]
### Description of the bug:
I'm not sure if this is a bug or feature, but for some reason `VirtualService` (a CRD of Istio, imported using `import` command) manifest doesn't get sorted:
```yaml
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
spec:
  gateways:
    - istio-system/api-gateway
  hosts:
    - my-host
  http:
      match:
        - uri:
            prefix: /
      route:
        - destination:
            host: my-service
            port:
              number: 8080
metadata:
  labels:
    app: my-service
    cdk8s.io/metadata.addr: my-service-deployment-c89b7b33
  name: my-service
  namespace: my-namespace
```

### Reproduction Steps:
Either import istio CRDs or use imported [networking.istio.io.ts.txt](https://github.com/cdk8s-team/cdk8s/files/10248640/networking.istio.io.ts.txt) (change the extension) and create a resource and then run `synth`. I expect the metadata to be before spec. Right?

### Environment:
  - **Framework Version**:  `2.3.0`
  - **OS**: maOS 12.6

---

This is :bug: Bug Report

**Comment by sumupitchayan** [comment]
This bug was fixed in a [recent PR](https://github.com/cdk8s-team/cdk8s-cli/pull/324) that ensures `metadata` is placed first after `apiVersion` and `kind`. Updating cdk8s should fix this issue.

However, there are still errors with the[ istio CRD yaml](https://github.com/istio/istio/blob/master/manifests/charts/base/crds/crd-all.gen.yaml). When importing the istio CRD via `cdk8s import`, I ran into this error when trying to compile the project:

<img width="1062" alt="Screenshot 2023-03-02 at 2 57 45 PM" src="https://user-images.githubusercontent.com/35242245/222541907-06c1508a-ce43-4a55-8079-ae365ce90116.png">

Digging further into the istio CRD yaml, we see that there are two variable names that cause this error: one named `mirror_percent` and `mirrorPercent` -- cdk8s will convert both of them to `mirrorPercent` since underscores like that are not used in variable names in Typescript. This is an issue that istio will have to fix in their CRD, since cdk8s uses its yaml manifest as the source of truth.
